Added support for minimumversion, maximumversion and requiredversion

gustavdw opened the pull request.

Commit:
Added support for RequiredVersion, MinimumVersion, MaximumVersion

Commit:
Fixed incorrect method name, and switch splatting

Commit:
Fixed issue with returning from foreach-object

Commit:
Updating moduleversion number

Commit:
Updated readme with instructions

Commit:
Variable name in testmethod was semantically incorrect

Commit:
Added installscope

Commit:
Corrected propertyname in readme

gustavdw commented:
I've added the installscope as well.

davidobrien1985 commented on PowerShellModule/PowerShellModule.psm1:
```diff
@@ -25,58 +41,122 @@ class PSModuleResource {
+            $arguments = $this.GetVersionArguments()
+            $arguments.Add("-Name", $this.Module_Name)
+            $arguments.Add("-Force", $true)
+            Uninstall-Module @arguments
```
Why not write the arguments as a hashtable?

gustavdw replied:
Refactored.

davidobrien1985 commented on PowerShellModule/PowerShellModule.psm1:
```diff
@@ -25,58 +41,122 @@ class PSModuleResource {
+        $returnVal = $false
+
+        Write-Verbose "The following versions of $($this.Module_Name) are installed"
+        $modules | % {
```
Would rather see a `foreach` here

gustavdw replied:
Fixed!

davidobrien1985 commented on PowerShellModule/Readme.md:
`````diff
@@ -24,6 +24,36 @@ Via Invoke-DscResource:
+````
 
 ## More Info:
 Read http://www.david-obrien.net/?p=3108
`````
Forgot to update this myself...
Mind changing this to `https://david-obrien.net/2015/09/powershell-dsc-to-manage-powershell-modules/` ? :)

gustavdw replied:
Yepp!

davidobrien1985 reviewed (commented):
Just minor things.
Thanks for contributing!!!

Commit:
Refactored usage of hashtables

Commit:
Changed verbose output of module versions to use foreach loop

Commit:
Updated link in readme

gustavdw commented:
@nzspambot will sort out next week.

Commit:
Added repository property. Updates to readme

Commit:
Refactoring naming of properties

Commit:
Refactoring

Commit:
Repository source prop is required. Ensure property now has default value, no longer req

Commit:
Checking for non-mandatory prop. Splatting.

Commit:
Incorrect property name. Added export of new resource

Commit:
Adding some verbose output for new resource

Commit:
Bugfix

gustavdw commented:
@nzspambot @davidobrien1985 
Added support for custom repositories (tested against a custom repository/gallery in ProGet with a connector to powershellgallery.com). It's implemented as a seperate resource which you can depend on from the psmoduleresource. Also updated the instructions/readme to new format with breakdown of resources and properties, with examples listed below. Set some default values in both resources for Ensure ('present').

Let me know if you have any feedback!

Commit:
Refactoring in readme

Commit:
Added some additional information to Get method of repositoryresource
